pkg/azure/credentials/azurecredentials.go

```diff
@@ -49,25 +44,11 @@ func FromSecret(ctx context.Context, client ctrl.Client, clusterID string) (*Ser
 	return &sp, nil
 }
 
-func findClusterCR(ctx context.Context, client ctrl.Client, clusterID string) (*capi.Cluster, error) {
-	clusterList := &capi.ClusterList{}
-	err := client.List(ctx, clusterList, ctrl.MatchingLabels{capi.ClusterLabelName: clusterID})
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-
-	if len(clusterList.Items) != 1 {
-		return nil, microerror.Maskf(executionFailedError, "Unable to find ClusterCR with label %q = %q. Expected 1 result, %d found.", capi.ClusterLabelName, clusterID, len(clusterList.Items))
-	}
-
-	return &clusterList.Items[0], nil
-}
-
 func findSecret(ctx context.Context, client ctrl.Client, orgName string) (*corev1.Secret, error) {
-	// Look for a secret with labels "app: credentiald" and "giantswarm.io/organization: <normalized org name>"
+	// Look for a secret with labels "app: credentiald" and "giantswarm.io/organization: org"
 	secrets := &corev1.SecretList{}
 
-	err := client.List(ctx, secrets, ctrl.MatchingLabels{"app": "credentiald", label.Organization: asDNSLabelName(orgName)})
```

Comment:
After that hassle we ended up with conclusion that organization label can not be normalized and therefore it cannot be for label selector either.